pkgs/build-support/writers/default.nix

```diff
@@ -94,8 +94,8 @@ rec {
         ]}
         gcc \
             ${optionalString (libraries != [])
-              "$(pkgs.pkgconfig}/bin/pkg-config --cflags --libs ${
-                concatMapStringsSep " " (lib: escapeShellArg (builtins.parseDrvName lib.name).name) (libraries)
+              "$(pkg-config --cflags --libs ${
+                concatMapStringsSep " " (pkg: "$(find ${escapeShellArg pkg}/lib/pkgsconfig -name \*.pc -exec basename {} \;)") libraries
```

Comment:
Should this not be `pkgconfig` instead of `pkgsconfig`?

Reply:
yes.

Reply:
argh, @Lassulus, did you even test it? :)

Reply:
I thought I did, but seems I made some mistake :/
Nevertheless, it would be very cool, if we had automated tests for writeC with libraries. Since I'm a C noob, I couldn't come up with a trivial example for a C program with a library.

Reply:
You can use this one: https://wiki.archlinux.org/index.php/Desktop_notifications#C
Only needs `glib` in buildInputs.

Reply:
"\*" and "\;" are wrong, `\` just goes away, there must be \\